components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -1274,12 +1274,16 @@ private boolean isDefaultGiven(String swaggerContent) throws APIManagementExcept
      */
     @Override
     public String processOtherSchemeScopes(String swaggerContent) throws APIManagementException {
-        if (!isDefaultGiven(swaggerContent)) {
-            Swagger swagger = getSwagger(swaggerContent);
-            swagger = processLegacyScopes(swagger);
+        Swagger swagger = getSwagger(swaggerContent);
+        Set<Scope> legacyScopes = getScopesFromExtensions(swagger);
+
+        if (!isDefaultGiven(swaggerContent) && legacyScopes.isEmpty()) {
```

Comment:
does legacyScopes will not null

Reply:
No. it is empty LinkedHashSet returning from getScopesFromExtensions

Reply:
https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java#L421

Reply:
ok